Example/Tests/Tests.m

```diff
@@ -10,39 +10,211 @@
+                                         };
+            
+            ICMUserAttributes *userAttributes = [ICMUserAttributes new];
+            userAttributes.companies = @[company];
+            
+            [verify(mockIntercom) updateUser:userAttributes];
+        });
     });
     
-    it(@"can read", ^{
-        expect(@"team").toNot.contain(@"I");
+    describe(@"track unknown users", ^{
+        beforeEach(^{
+            SEGIdentifyPayload *identifyPayload = [[SEGIdentifyPayload alloc] initWithUserId:nil anonymousId:@"324908523402" traits:@{
+                                                                                                                                       @"gender" : @"female",
+                                                                                                                                       @"company" : @"segment",
+                                                                                                                                       @"name" : @"ladan"
+                                                                                                                                       } context:@{}
+                                                                                integrations:@{}];
+            [integration identify:identifyPayload];
+            [verify(mockIntercom) registerUnidentifiedUser];
+        });
+        
+        it(@"calls track with properties", ^{
+            NSDictionary *properties = @{
+                                         @"name" : @"Bob",
+                                         @"gender" : @"male"
+                                         };
+            SEGTrackPayload *payload = [[SEGTrackPayload alloc] initWithEvent:@"Event" properties:properties context:@{
+                                                                                                                       } integrations:@{}];
+            
+            [integration track:payload];
+        });
+        
+        it(@"calls track without properties", ^{
+            SEGTrackPayload *payload = [[SEGTrackPayload alloc] initWithEvent:@"Event" properties:@{} context:@{
+                                                                                                                       } integrations:@{}];
+            
+            [integration track:payload];
+            [verify(mockIntercom) logEventWithName:@"Event"];
+        });
     });
     
-    it(@"will wait and succeed", ^{
-        waitUntil(^(DoneCallback done) {
-            done();
+    describe(@"identify", ^{
+        it(@"identifies an unknown user with traits", ^{
+            SEGIdentifyPayload *identifyPayload = [[SEGIdentifyPayload alloc] initWithUserId:nil anonymousId:@"324908523402" traits:@{
+                                                                                                                                      @"gender" : @"female",
```

Comment:
formatting seems really inconsistent, are you using space commander for all?

Reply:
yeah - I can re-run and see whats up